habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/ContainerizeDependenciesMojo.java

```diff
@@ -0,0 +1,156 @@
+package org.technologybrewery.habushu;
+
+import org.apache.maven.execution.MavenSession;
+import org.apache.maven.model.Dependency;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Component;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.project.MavenProject;
+import org.apache.maven.shared.model.fileset.FileSet;
+import org.apache.maven.shared.model.fileset.util.FileSetManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Stages the source files of a monorepo dependency
+ * to the target directory along with the source files of
+ * any transitive path-based dependencies.
+ */
+
+@Mojo(name = "containerize-dependencies", defaultPhase = LifecyclePhase.VALIDATE)
+public class ContainerizeDependenciesMojo extends AbstractHabushuMojo {
+
+    private static final Logger logger = LoggerFactory.getLogger(ContainerizeDependenciesMojo.class);
+
+    @Component
+    protected MavenSession session;
+
+    protected File anchorSourceDirectory;
+    protected File anchorOutputDirectory;
+    protected final String HABUSHU = "habushu";
+
+    /**
+     * Overriding to allow execution in non-habushu projects.
+     */
+    @Override
+    public void execute() throws MojoExecutionException, MojoFailureException {
+        doExecute();
+    }
+
+    @Override
+    protected void doExecute() throws MojoExecutionException, MojoFailureException {
+        if (this.anchorSourceDirectory == null) {
+            this.anchorSourceDirectory = new File(session.getExecutionRootDirectory());
```

Comment:
A: Why make this a field on the class? Same with `anchorOutputDirectory`.  I would think we could just pass it into the method call.

Reply:
The anchor attributes were important for setting up the tests. Tried pulling it out, but ended up breaking the tests. For the sake of time, I'm going to keep this as is, but when we clean up the test case in the follow-on, I think we can implement what you're suggesting

Reply:
Update: these are now configurable parameters